tuna/rocmlir/rocmlir_worker.py

```diff
@@ -0,0 +1,165 @@
+
+
+class RocMLIRWorker(WorkerInterface):
+  """ The RocMLIR class implements the worker class. Its purpose is to run a command. It picks up
+  new jobs and when completed, sets the state to completed. """
+
+  def __init__(self, **kwargs):
+    """Constructor"""
+    self.dbt = None
+    super().__init__(**kwargs)
+    self.set_db_tables()
+    self.result_attr = [column.name for column in inspect(self.dbt.results).c]
+    self.result_attr.remove("insert_ts")
+    self.result_attr.remove("update_ts")
+
+  def set_db_tables(self):
+    """Initialize tables"""
+    self.dbt = RocMLIRDBTables(session_id=self.session_id)
+
+  def update_result_table(self, session, result_str):
+    """update results table with individual result entry"""
+    obj = self.dbt.results()
+
+    arch, config, perf_config, tflops = obj.parse(result_str)
+
+    print(f"arch = '{arch}', config = '{config}', perf_config = '{perf_config}', tflops = {tflops}",
+          file=sys.stderr)
+
+    obj.valid = 1
+    obj.session = self.dbt.session.id
+    obj.arch = arch
+    obj.config = self.job.config
+    obj.config_str = config
+    obj.perf_config = perf_config
+    obj.kernel_tflops = tflops
+
+    self.logger.info('Inserting results for job_id=%s', self.job.id)
+    query = gen_insert_query(obj, self.result_attr,
+                             self.dbt.results.__tablename__)
+    session.execute(query)
+    session.commit()
+    return True
+
+  def process_result(self, result_str: str):
+    """process tuning-run results"""
+    status = []
+    with DbSession() as session:
+
+      def actuator(func, result_str):
+        return func(session, result_str)
+
+      #retry returns false on failure, callback return on success
+      ret = session_retry(session, self.update_result_table,
+                          functools.partial(actuator, result_str=result_str),
+                          self.logger)
+      if not ret:
+        self.logger.warning('RocMLIR:  Unable to update database')
+
+    return status
+
+  def output_filename(self):
+    """Canonical name for tuningRunner.py output for a job."""
+    return f"tuning-results-{self.job.id}.tsv"
+
+  def step(self):
+    """Main functionality of the worker class. It picks up jobs in new state and executes them"""
+
+    if not self.get_job("new", "running", False):
+      #Sleep in case of DB contention
+      sleep(random.randint(1, 10))
+      return False
+
+    self.logger.info('Acquired new job: job_id=%s', self.job.id)
+    self.set_job_state('running')
+
+    try:
+      retcode,cmd_output = self.run_cmd()
+    except ValueError as verr:
+      self.logger.info(verr)
+      self.set_job_state('errored', result=verr)
+    else:
+      if retcode != 0:
+        msg = f"Error code {retcode}, output {cmd_output}"
+        self.logger.info(msg)
+        self.set_job_state('errored', result=msg)
+      else:
+        try:
+          with open(self.output_filename(), 'r', encoding='utf8') as results:
```

Comment:
tuningRunner.py writes an output file with the winning perf-config (or more than one, if given more than one conv-config, which we aren't doing here).  I haven't managed to make it output to stdout instead and thus have to open-and-read the file, and also worry about file-not-found when something weird happens and deleting the file to avoid name collisions when I reinitialise counters and flush databases.

I have hopes of cleaning this up.